code/code/misc/monster.cc

```diff
@@ -513,45 +513,37 @@ int TMonster::getMobDamage() const
   return max(1, idamrnd);
 }
 
-int TMonster::lookForEngaged(const TBeing *ch)
-{
-  // here's some AI
-  // tank and healer fight mob, tank flees, mob should now beat on
-  // healer (regardless of whether he engaged or not?)
-  if (isPc())
-    return FALSE;
+int TMonster::lookForEngaged(const TBeing *previousAttacker) {
+  if (isPc()) return false;
+
   // if someone is attacking me directly, bypass
-  if (attackers > 0)
-    return FALSE;
-  if (!isSmartMob(0))
-    return FALSE;
-
-  TThing * t;
-  for(StuffIter it=roomp->stuff.begin();it!=roomp->stuff.end();){
-    t=*(it++);
-    if (t == this || (ch && dynamic_cast<const TBeing *>(t) == ch))
-      continue;
-    TBeing *tbt = dynamic_cast<TBeing *>(t);
-    if (!tbt)
+  if (attackers > 0) return false;
+
+  for (auto thing : roomp->stuff) {
+    if (thing == this ||
+        (previousAttacker && dynamic_cast<const TBeing *>(thing) == previousAttacker))
       continue;
-    if (tbt->fight() == this) {
-      // I'm a smart mob, and, oh look, tbt is engaged with me...
-      int rc = takeFirstHit(*tbt);
+
+    auto thingAsTBeing = dynamic_cast<TBeing *>(thing);
```

Comment:
Perhaps we can avoid the double cast by using `thingAsTBeing` on line  524

Reply:
Resolved in https://github.com/sneezymud/sneezymud/pull/373/commits/a9d0e2528b68faa35f2ebc84115e123950d13d6f